bin/ofs/src/fuse.rs

```diff
@@ -401,26 +386,39 @@ impl PathFilesystem for Fuse {
             flags
         );
 
-        let (is_read, is_write) = self.check_flags(flags)?;
+        let (is_read, is_trunc, is_append) = self.check_flags(flags | libc::O_CREAT as u32)?;
 
         let path = PathBuf::from(parent).join(name);
         self.op
             .write(&path.to_string_lossy(), Bytes::new())
             .await
             .map_err(opendal_error2errno)?;
 
-        let metadata = Metadata::new(EntryMode::FILE);
-        let attr = metadata2file_attr(&metadata, SystemTime::now(), self.uid, self.gid);
+        let inner_writer = if is_trunc || is_append {
+            let writer = self
+                .op
+                .writer_with(&path.to_string_lossy())
+                .append(is_append)
+                .await
+                .map_err(opendal_error2errno)?;
+            Some(Arc::new(Mutex::new(InnerWriter { writer, written: 0 })))
+        } else {
+            None
+        };
+
+        let now = Utc::now();
+        let metadata = Metadata::new(EntryMode::FILE)
```

Comment:
Please call stat to fetch metadata. It's invalid to build metadata outside of services.

Maybe we should build fs's own metadata instead of using `opendal::Metadata` here?

Reply:
Maybe we can fix it later